src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp

```diff
@@ -280,9 +280,9 @@ struct typed_primitive_impl_ocl : public typed_primitive_impl<PType> {
                                    << (needs_completion_event ? " has_completion_event=true" : "") << std::endl;
 
             auto ev = stream.enqueue_kernel(*_kernels[kd_idx], params, args, tmp_events, needs_completion_event);
-            new_events.push_back(ev);
+            if (_kernel_data.needs_sub_kernels_sync)
+                new_events.push_back(ev);
             all_events.push_back(ev);
-
```

Comment:
Looks like it may break pipeline for events synchronization type
Let's assume that current Concat has N kernels and depends on N input events
Before:
1) Concat_0 kernel uses `tmp_events(events)` as dependency events, adds its' output event to `new_events` and saves it in `tmp_events` 
2) Concat_1 kernel uses `tmp_events` (which includes Concat_0's kernel dependency, which depends on 3 input events), so Concat_1 have to be executed only after Concat_0 (and 3 original input dependency respectively)
...

After: 
1) Concat_0 kernel uses `tmp_events(events)` as dependency events, but doesn't add its output event to `new_events` (because of needs_sub_kernels_sync) and copies empty vector to `tmp_events` 
2) Concat_1 kernel uses empty `tmp_events` vector and can be executed w/o any dependencies at all

So, in case of `needs_sub_kernels_sync=false` we need to pass original dependencies to `stream.enqueue_kernel()` for the second and next sub-kernels

Reply:
Thanks for your comment, I agree your point, but the reason why it is still okay with as is, the barrier is already injected in the first kernel. But for safety I will update the depdencies. :)

Reply:
Yeah, for barriers sync it was completely okay. My point was mainly about `sync_methods::events` synchronization type, but now it looks good, thank you! :)